components/formatted_markdown_message.jsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedHTMLMessage, injectIntl, intlShape} from 'react-intl';
+import PropTypes from 'prop-types';
+import marked from 'marked';
+
+const UNESCAPED_TAGS = ['BR'];
+const TARGET_BLANK_URL_PREFIX = '!';
+
+class FormattedMarkdownMessage extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.renderer = new marked.Renderer();
+        this.renderer.link = FormattedMarkdownMessage.linkF;
+    }
+
+    static linkF(href, title, text) {
+        if (href[0] === TARGET_BLANK_URL_PREFIX) {
+            return `<a href="${href.substring(1, href.length)}" target="_blank">${text}</a>`;
+        }
+        return `<a href="${href}">${text}</a>`;
+    }
+
+    static get propTypes() {
+        return {
+            intl: intlShape.isRequired,
+            id: PropTypes.string.isRequired,
+            defaultMessage: PropTypes.string.isRequired,
+            values: PropTypes.object,
+        };
+    }
+
+    static escapeHtml(str) {
+        var div = document.createElement('div');
+        div.appendChild(document.createTextNode(str));
+        return div.innerHTML;
+    }
+
+    static sanitizer(p1) {
```

Comment:
One other thought about this part, do we even need to support `<br/>` being included in the translation string? Why not just use regular markdown line breaks? That would get rid of the need to sanitize it manually since marked would handle that itself.

Reply:
That was the first thing I tried but couldn't get the GFM line breaks working... even with the `breaks` option set to true. Let me revisit and see if I made a mistake in my first attempt because I agree that it would be nice to remove the `<br>`s too.

Reply:
Maybe take a look at the options we pass into the renderer in `utils/markdown/index.js` since I think those might change the behaviour of line endings.